pkg/ingester/ingester.go

```diff
@@ -1515,13 +1612,24 @@ func (i *Ingester) queryStreamSamples(ctx context.Context, db *userTSDB, from, t
 
 		it = series.Iterator(it)
 		for valType := it.Next(); valType != chunkenc.ValNone; valType = it.Next() {
-			if valType != chunkenc.ValFloat {
-				return 0, 0, fmt.Errorf("unsupported value type: %v", valType)
+			switch valType {
+			case chunkenc.ValFloat:
+				t, v := it.At()
+				ts.Samples = append(ts.Samples, mimirpb.Sample{Value: v, TimestampMs: t})
+			case chunkenc.ValHistogram:
+				// TODO when read path for native histograms is ready, return Histogram samples
+				// t, v := it.AtHistogram()
+				// ts.Histograms = append(ts.Histograms, mimirpb.FromHistogramToHistogramProto(t, v))
+			case chunkenc.ValFloatHistogram:
+				// TODO when read path for native histograms is ready, return Histogram samples
+				// t, v := it.AtFloatHistogram()
+				// ts.Histograms = append(ts.Histograms, mimirpb.FromFloatHistogramToHistogramProto(t, v))
+			default:
+				return 0, 0, 0, fmt.Errorf("unsupported value type: %v", valType)
 			}
```

Comment:
Please remove commented code. It's OK to either ignore histogram samples for now, or return error (ie. do no change to read path in this PR).

Reply:
done, plus created a reminder, just in case https://github.com/grafana/mimir/issues/4176